Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1674,6 +1678,116 @@ JS::GCPtr<CSSSupportsRule> Parser::convert_to_supports_rule(Rule& rule)
+        return {};
+    }
+
+    auto const& name = name_token.ident();
+
+    auto block_tokens = rule.block()->values();
+    auto block_stream = TokenStream { block_tokens };
+
+    auto list_of_declarations = parse_a_list_of_declarations(block_stream);
+
+    auto syntax_maybe = Optional<FlyString> {};
+    auto inherit_maybe = Optional<bool> {};
+    auto initial_value_maybe = Optional<FlyString> {};
+
+    for (auto const& declOrRule : list_of_declarations) {
+        if (!declOrRule.is_declaration()) {
+            dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Only declarations allowed in @property block.; discarding.");
+            return {};
+        }
+        auto const& decl = declOrRule.declaration();
+
+        if (decl.name() == "syntax") {
+            if (syntax_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"syntax\" in @property block; discarding.");
+                return {};
+            }
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::String)) {
+                syntax_maybe = decl.values()[0].token().string();
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"syntax\": {}; discarding.", decl.to_string());
+            }
+        }
+        if (decl.name() == "inherits") {
+            if (inherit_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"inherits\" in @property block; discarding.");
+                return {};
+            }
+            // Feels hacky using Ident here, but this is what the tokenizer gives us.
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::Ident)) {
+                auto const& ident = decl.values()[0].token().ident();
+                if (ident.is_one_of("true", "false")) {
+                    inherit_maybe = (ident == "true");
+                } else {
+                    dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Expected true/false for @property \"inherits\" value, got: {}; discarding.", ident.to_string());
+                }
+
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"inherits\": {}; discarding.", decl.to_string());
+            }
+        }
+        if (decl.name() == "initial-value") {
+            if (initial_value_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"initalValue\" in @property block; discarding.");
+                return {};
+            }
+            // FIXME: Ensure that the initial value matches the syntax, and parse the correct CSSValue out
+            StringBuilder initial_value_sb;
+            for (auto const& component : decl.values()) {
+                initial_value_sb.append(component.to_string());
+            }
+            initial_value_maybe = MUST(initial_value_sb.to_fly_string());
```

Comment:
This could be a regular `String` rather than a `FlyString`. It's only used in one place and nobody ever compares it.

Reply:
Should the same be done for `syntax`? I feel yea, but not sure how FlyString works with tokenizing a string (which we'll do in the future). 

From my understanding of the Flywheel pattern (took me a while to get that this was what Fly stood for in FlyString), it'd only help with full string to string comparisons, so I guess it should be changed to `String`.

Reply:
Flyweight pattern, but yes. I keep thinking we have documentation about the string types and then realising we don't. :sweat_smile: 

For `syntax`... we already have a `FlyString` from the CSS string token, so you don't really gain anything from using a `String` instead, even though you're right and we don't need to compare it.